### PR TITLE
YETUS-1233. 2023-09 Dependency Updates

### PR DESCRIPTION
**Pull request by aw-was-here**

### CODE REVIEW

File: .linkcheckerrc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[filtering]
+ignore=
+  http://localhost:8123/documentation/[0-9]+.[0-9]+.[0-9]+
```
Comment (ndimiduk) [inline .linkcheckerrc:18]: Why this ignore line? We don't want to check broken links on built documentation?

File: asf-site-src/pom.xml [hunk 2]
```diff
@@ -84,7 +84,6 @@
             </goals>
             <configuration>
               <executable>${rdm.bin}</executable>
-              <requiresOnline>true</requiresOnline>
```
Comment (ndimiduk) [inline asf-site-src/pom.xml:87]: oh why were these here in the first place?

File: precommit/src/main/shell/test-patch-docker/Dockerfile [hunk 3]
```diff
@@ -149,16 +149,22 @@ FROM yetusbase AS yetushadolint
 ARG HADOLINT_VERSION=2.12.0
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 RUN if [[ "$(uname -m)" == "x86_64" ]]; then curl -sSL \
-        https://github.com/hadolint/hadolint/releases/download/v$HADOLINT_VERSION/hadolint-Linux-"$(uname -m)" \
+        https://github.com/hadolint/hadolint/releases/download/v$HADOLINT_VERSION/hadolint-Linux-x86_64 \
```
Comment (ndimiduk) [inline precommit/src/main/shell/test-patch-docker/Dockerfile:152]: I still prefer the style I proposed in my patch because docker can cache the downloaded artifacts between runs. The layer caching speeds up rebuild cycles drastically.

**Review by ndimiduk — APPROVED** [review]
Thanks for this cleanup.

File: .linkcheckerrc (hunk 1 above)
Comment (aw-was-here) [reply .linkcheckerrc:18]: It is specifically ignoring the references to previous release versions of the documentation which don't get built as part of this process but do sit on the web server.

File: asf-site-src/pom.xml (hunk 2 above)
Comment (aw-was-here) [reply asf-site-src/pom.xml:87]: At some point in time, we either used a plugin that was offline/online aware or they removed the functionality from maven.  But they were there so that you could tell maven you were offline and it wouldn't try to connect to JIRA to generate the release notes, etc.
